SampleGallery/Samples/SDK 17134/RadialGradients/RadialGradients.xaml.cs

```diff
@@ -0,0 +1,310 @@
+            _buttonVisual.AnchorPoint = new Vector2(.5f, .5f);
+            _buttonVisual.Brush = _buttonBrush;
+
+            // Create a geometric clip that clips the rectangular sprite visual to a circle shape
+            CompositionGeometricClip gClip = _compositor.CreateGeometricClip();
+            CompositionEllipseGeometry circle = _compositor.CreateEllipseGeometry();
+            circle.Radius = new Vector2(_buttonVisual.Size.X / 2, _buttonVisual.Size.Y / 2);
+            circle.Center = new Vector2(_buttonVisual.Size.X / 2, _buttonVisual.Size.Y / 2);
+            gClip.Geometry = circle;
+
+            _buttonVisual.Clip = gClip;
+
+            // Create the brush for the backing visual 
+            _pulseBrush = _compositor.CreateRadialGradientBrush();
+            _pulseBrush.EllipseCenter = new Vector2(.5f, .5f);
+            _pulseBrush.EllipseRadius = new Vector2(.5f, .5f);
+
+            _PBGradientStop1 = _compositor.CreateColorGradientStop();
+            _PBGradientStop1.Offset = 0;
+            _PBGradientStop1.Color = _transparent;
+            _PBGradientStop2 = _compositor.CreateColorGradientStop();
+            _PBGradientStop2.Offset = 1;
+            _PBGradientStop2.Color = _transparent;
+
+            _pulseBrush.ColorStops.Add(_PBGradientStop1);
+            _pulseBrush.ColorStops.Add(_PBGradientStop2);
+
+            // Finish defining the properties of the backing visual creates a pulsing effect when animated
+            _pulseVisual.Size = new Vector2(500, 500);
+            _pulseVisual.Offset = new Vector3(((float)Rectangle1.ActualWidth / 2), ((float)Rectangle1.ActualHeight / 2), 0);
+            _pulseVisual.AnchorPoint = new Vector2(.5f, .5f);
+            _pulseVisual.Brush = _pulseBrush;
+
+            // Create a geometric clip that clips the rectangular sprite visual to a circle shape
+            CompositionGeometricClip gClip2 = _compositor.CreateGeometricClip();
+            CompositionEllipseGeometry circle2 = _compositor.CreateEllipseGeometry();
+            circle2.Radius = new Vector2(_pulseVisual.Size.X / 2, _pulseVisual.Size.Y / 2);
+            circle2.Center = new Vector2(_pulseVisual.Size.X / 2, _pulseVisual.Size.Y / 2);
+            gClip2.Geometry = circle2;
+
+            _pulseVisual.Clip = gClip2;
+
+            // Tie sprite visuals to corresponding XAML UI Elements
+            ElementCompositionPreview.SetElementChildVisual(Rectangle1, _pulseVisual);
+            ElementCompositionPreview.SetElementChildVisual(Rectangle2, _buttonVisual);
+
+            // WHen the page first loads we want the animations to be off so we set this boolean flag to false
+            animationOn = false;
+        }
+
+        // When the XAML element is clicked, this method with kick off the animation of the stop watch or turn it back to it's original state depending on if it's on or off
+        private void OnClick1(object sender, RoutedEventArgs e)
```

Comment:
Consider using a better name so that the comment doesn't need to explain so much.